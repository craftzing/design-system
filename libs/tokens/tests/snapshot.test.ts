import { readFile } from 'fs/promises';
import path from 'path';
import { expect, describe, it } from 'vitest';

const BUILD_PATH = './dist';
const formats = ['css'];

describe('Token Build Snapshot Tests', () => {
  it.each(
    formats.map((format) => ({
      format,
      fileExtension: format,
    }))
  )(
    'should match the snapshot for $tokenType $format tokens',
    async ({ format, fileExtension }) => {
      const tokenFilePath = path.join(
        BUILD_PATH,
        format,
        `tokens.${fileExtension}`
      );
      const tokenFileContent = await readFile(tokenFilePath, 'utf-8');
      expect(tokenFileContent).toMatchSnapshot();
    }
  );
});
