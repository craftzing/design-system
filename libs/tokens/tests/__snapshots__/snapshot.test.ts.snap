// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Token Build Snapshot Tests > should match the snapshot for undefined 'css' tokens 1`] = `
":root {
  --400: 400;
  --500: 500;
  --600: 600;
  --700: 700;
  --purple-100: #f3f1fb;
  --purple-200: #e4dff6;
  --purple-300: #d5cdf2;
  --purple-400: #9887de;
  --purple-500: #7a64d5;
  --purple-600: #6b52d0;
  --purple-700: #513bac;
  --purple-800: #3f317b;
  --purple-900: #2a2051;
  --purple-1000: #1e173a;
  --gray-50: #fafafa;
  --gray-100: #f5f5f5;
  --gray-200: #ececed;
  --gray-300: #d5d5d8;
  --gray-400: #bebdc2;
  --gray-500: #a7a6ac;
  --gray-600: #706d78;
  --gray-700: #4b4855;
  --gray-750: #201c2b;
  --gray-800: #1d1929;
  --gray-900: #161320;
  --white: #ffffff;
  --red-50: #fef2f2;
  --red-100: #fee2e2;
  --red-200: #fecaca;
  --red-300: #fba6a6;
  --red-400: #f77272;
  --red-500: #de3535;
  --red-600: #da2828;
  --red-700: #b81d1d;
  --red-800: #981c1c;
  --red-900: #7e1e1e;
  --space-4: 0.25rem;
  --red-950: #440b0b;
  --space-8: 0.5rem;
  --space-12: 0.75rem;
  --space-16: 1rem;
  --space-20: 1.25rem;
  --space-24: 1.5rem;
  --space-28: 1.75rem;
  --space-32: 2rem;
  --space-40: 2.5rem;
  --space-48: 3rem;
  --space-56: 3.5rem;
  --space-64: 4rem;
  --space-72: 4.5rem;
  --space-80: 5rem;
  --space-88: 5.5rem;
  --space-96: 6rem;
  --space-104: 6.5rem;
  --space-128: 8rem;
  --space-160: 10rem;
  --size-12: 0.75rem;
  --size-14: 0.875rem;
  --size-16: 1rem;
  --size-18: 1.125rem;
  --size-20: 1.25rem;
  --size-24: 1.5rem;
  --size-28: 1.75rem;
  --size-32: 2rem;
  --size-40: 2.5rem;
  --size-60: 3.75rem;
  --line-height-160: 160%;
  --line-height-140: 140%;
  --primary-100: var(--purple-100);
  --primary-200: var(--purple-200);
  --primary-300: var(--purple-300);
  --primary-400: var(--purple-400);
  --primary-500: var(--purple-500);
  --primary-600: var(--purple-600);
  --primary-700: var(--purple-700);
  --primary-800: var(--purple-800);
  --primary-900: var(--purple-900);
  --primary-1000: var(--purple-1000);
  --neutral-50: var(--gray-50);
  --neutral-100: var(--gray-100);
  --neutral-200: var(--gray-200);
  --neutral-300: var(--gray-300);
  --neutral-400: var(--gray-400);
  --neutral-500: var(--gray-500);
  --neutral-600: var(--gray-600);
  --neutral-700: var(--gray-700);
  --neutral-750: var(--gray-750);
  --neutral-800: var(--gray-800);
  --neutral-900: var(--gray-900);
  --body-size-s: var(--size-14);
  --body-size-m: var(--size-16);
  --body-size-l: var(--size-18);
  --body-base: var(--400) var(--body-size-m)/var(--line-height-160) sans-serif;
  --heading-size-xxs: var(--size-16);
  --heading-size-xs: var(--size-20);
  --heading-size-s: var(--size-24);
  --heading-size-m: var(--size-28);
  --heading-size-l: var(--size-32);
  --heading-size-xl: var(--size-40);
  --heading-h1: var(--700) var(--heading-size-xl)/var(--line-height-140) sans-serif;
  --heading-h2: var(--700) var(--heading-size-l)/var(--line-height-140) sans-serif;
  --heading-h3: var(--600) var(--heading-size-m)/var(--line-height-140) sans-serif;
  --error-50: var(--red-50);
  --error-100: var(--red-100);
  --error-200: var(--red-200);
  --error-300: var(--red-300);
  --error-400: var(--red-400);
  --error-500: var(--red-500);
  --error-600: var(--red-600);
  --error-700: var(--red-700);
  --error-800: var(--red-800);
  --error-900: var(--red-900);
  --error-950: var(--red-950);
  --typography-black: var(--gray-900);
  --typography-white: var(--white);
  --regular: var(--400);
  --medium: var(--500);
  --semibold: var(--600);
  --bold: var(--700);
  --space-xxxs: var(--space-4);
  --space-xxs: var(--space-8);
  --space-xs: var(--space-12);
  --space-s: var(--space-16);
  --space-m: var(--space-20);
  --space-l: var(--space-24);
  --space-xl: var(--space-28);
  --space-xxl: var(--space-32);
  --space-xxxl: var(--space-40);
  --size-xs: var(--size-12);
  --size-s: var(--size-14);
  --size-m: var(--size-16);
  --size-l: var(--size-18);
  --size-xl: var(--size-20);
}
"
`;
